desktop/appcontainer/Message.js

```diff
@@ -53,32 +55,37 @@ export class Message extends Component {
     }
 
     renderInput() {
-        const {formModel, input} = this.model;
+        const {formModel, input, autoFocus} = this.model;
         if (!formModel) return null;
         return form({
             model: formModel,
             fieldDefaults: {commitOnChange: true, minimal: true, label: null},
             item: formField({
                 field: 'value',
-                item: withDefault(input.item, textInput({autoFocus: true}))
+                item: withDefault(input.item, textInput({
+                    autoFocus:  autoFocus == 'input' ? true : false
+                }))
             })
         });
     }
 
     renderButtons() {
-        const {formModel, confirmText, cancelText, confirmIntent, cancelIntent} = this.model;
+        const {formModel, confirmText, cancelText, confirmIntent, cancelIntent, autoFocus} = this.model;
         return [
             filler(),
             button({
                 text: cancelText,
                 omit: !cancelText,
                 intent: cancelIntent,
+                autoFocus:  autoFocus == 'cancel' ? true : false,
                 onClick: () => this.model.doCancel()
             }),
             button({
                 text: confirmText,
                 intent: confirmIntent,
                 disabled: formModel ? !formModel.isValid : false,
+                autoFocus:  autoFocus == 'confirm' ? true : false,
+                // minimal: false,
```

Comment:
I too _sometimes_ want a non-minimal button here...  Makes me wonder if we should have made the config `confirmProps` instead of calling out text and intent...  (And used the same for cancel of course.)

If we did that, we could lose the autoFocus config for messages and simply pass in autoFocus to either button as a prop. Let's discuss quickly tomorrow to see if that feels worth it.  I think it might be - avoids the issue with input above + having the new config with string values vs. true/false as seen elsewhere.

Given that these are pretty commonly used methods, we could look for the dedicated text/intent configs - if found, log a deprecation warning before applying them into the new object-based configs automatically.  That would let us make this change while keeping things non-breaking